j5/boards/base.py

```diff
@@ -114,21 +113,16 @@ def __next__(self) -> Board:
         """Get the next item in the iteration."""
         if self._iterator_counter >= len(self.boards):
             raise StopIteration
-        board = self.boards[self._iterator_counter]
+        board = self.boards[list(self.boards.keys())[self._iterator_counter]]
         self._iterator_counter += 1
         return board
 
-    def __getitem__(self, index: BoardIndex) -> Board:
+    def __getitem__(self, serial: str):
         """Get the board from an index."""
         if len(self.boards) <= 0:
```

Comment:
Is it worth removing this check as well?

Reply:
It does seem a little redundant since the if statement on line 126 will cover that as well. Are there any situations where it would be useful to know that no boards are connected rather than the board not being found?

Reply:
@trickeydan can you think of any? My thoughts are that the object already has `__len__` in place, so we shouldn't _need_ to do that here. But if people decide they'd rather have it here as well, I'm fine with that too. And typically, `__getitem__` for python doesn't report the object is empty.

Reply:
So there are cases when the `BoardGroup` will be empty. However, as we are removing the ability to index by integer here, I would be fine with the removal of this check.